src/contract-ui/tabs/edit-extensions/components/ExtensionForm.tsx

```diff
@@ -0,0 +1,406 @@
+import { useForm } from "react-hook-form";
+import {
+  Flex,
+  Input,
+  Select,
+  FormControl,
+  Skeleton,
+  Spacer,
+  Grid,
+  useToast,
+  Spinner,
+} from "@chakra-ui/react";
+import { FormLabel, FormErrorMessage, Text } from "tw-components";
+import {
+  useAllVersions,
+  usePublishedContractsQuery,
+} from "components/contract-components/hooks";
+import { installPublishedExtension } from "thirdweb/extensions/modular";
+import {
+  sendTransaction,
+  waitForReceipt,
+  type ContractOptions,
+} from "thirdweb";
+import type { Account } from "thirdweb/wallets";
+import { useMutation, useQuery } from "@tanstack/react-query";
+import { thirdwebClient } from "../../../../@/constants/client";
+import { compatibleExtensions } from "@thirdweb-dev/sdk";
+import { download } from "thirdweb/storage";
+import { resolveImplementation } from "thirdweb/utils";
+import { TransactionButton } from "components/buttons/TransactionButton";
+
+type FormData = {
+  publisherAddress: string;
+  extensionContract: string;
+  version: string;
+};
+
+type InstallExtensionFormProps = {
+  contract: ContractOptions;
+  refetchExtensions: () => void;
+  account: Account;
+  installedExtensions: {
+    data?: string[];
+    isLoading: boolean;
+  };
+};
+
+export const InstallExtensionForm = (props: InstallExtensionFormProps) => {
+  const { register, watch, formState, resetField, reset } = useForm<FormData>({
+    defaultValues: {
+      version: "latest",
+    },
+  });
+  const toast = useToast();
+  const { contract, account } = props;
+  const { errors } = formState;
+
+  const publishedContractsQuery = usePublishedContractsQuery(
+    watch("publisherAddress"),
+    "ModularExtension",
+  );
+  const allVersions = useAllVersions(
+    watch("publisherAddress"),
+    watch("extensionContract"),
+  );
+
+  const installMutation = useMutation({
+    mutationFn: async () => {
+      const installTransaction = installPublishedExtension({
+        contract,
+        chain: contract.chain,
+        client: contract.client,
+        account,
+        extensionName: watch("extensionContract"),
+        publisherAddress: watch("publisherAddress"),
+        version: watch("version"),
+      });
+
+      const txResult = await sendTransaction({
+        transaction: installTransaction,
+        account,
+      });
+
+      await waitForReceipt(txResult);
+    },
+  });
+
+  const onSubmit = async () => {
+    try {
+      await installMutation.mutateAsync();
```

Comment:
this should use `installationMutation.mutate` and then do the async callbacks inside the `onSuccess` and `onError` of the mutation